docs/architecture/grpc-for-wcf-developers/protobuf-any-oneof.md

```diff
@@ -1,11 +1,105 @@
 ---
 title: Protobuf Any and OneOf fields for variant types - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: Using the Any type and OneOf keyword to represent variant object types in messages
 author: markrendle
-ms.date: 09/02/2019
+ms.date: 09/09/2019
 ---
 
 # Protobuf Any and OneOf fields for variant types
 
+Handling dynamic property types (i.e. properties of type `object`) in WCF is complicated. Serializers must be specified, `[KnownType]` attributes must be provided, and so on.
+
+Protobuf provides two simpler options for dealing with values that may be of more than one type. The `Any` type can represent any known Protobuf message type, while the `oneOf` keyword allows you to specify that only one of a range of fields can be set in any given message.
+
+## Any
+
+To use the any type, you must import `google/protobuf/any.proto`.
+
+```protobuf
+syntax "proto3"
+
+import "google/protobuf/any.proto"
+
+message Stock {
+    // Stock-specific data
+}
+
+message Currency {
+    // Currency-specific data
+}
+
+message ChangeNotification {
+    int32 id = 1;
+    google.protobuf.Any instrument = 2;
+}
+```
+
+In the C# code, the `Any` type provides methods for setting the field, extracting the message and checking the type.
+
+```csharp
+public void FormatChangeNotification(ChangeNotification change)
+{
+    if (change.Instrument.Is(Stock.Descriptor))
+    {
+        FormatStock(change.Instrument.Unpack<Stock>());
+    }
+    else if (change.Instrument.Is(Currency.Descriptor))
+    {
+        FormatCurrency(change.Instrument.Unpack<Currency>());
+    }
+    else
+    {
+        throw new ArgumentException("Unknown instrument type");
+    }
+}
+```
+
+The `Descriptor` static field on each generated type is used by Protobuf's internal reflection code to resolve `Any` field types. There is also a `TryUnpack<T>` method, but that creates an uninitialized instance of `T` even when it fails, so it's better to use the `Is` method as shown above.
+
+## OneOf
+
+Whereas `Any` is a well-known type, `oneof` is a Protobuf language keyword. Using `oneof` to specify the `ChangeNotification` message might look like this:
```

Comment:
whereas is considered too formal by our review tool. Do we really need to contrast these two concepts like this? Why is it important to know that one is a type and the other a keyword?